src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorShape.cs

```diff
@@ -987,7 +985,7 @@ public TensorShape Slice<TGetOffsetAndLength, T>(ReadOnlySpan<T> state, out nint
 
             TensorShape result = new TensorShape(
                 flattenedLength,
-                linearLength,
+                LinearLength - computedOffset,
```

Comment:
Is this correct in all scenarios?

In the main constructor loop I've got it computing a `minimumLinearLength` based on the `stride * length` (where-as flattenedLength is `flattenedLength *= stride`).

I think it would be good to normalize to a similar pattern between the two, to help avoid any bugs or edge cases.